File: java/sample-springmvc-api/src/main/java/io/github/rajendrasatpute/samplespringmvcapi/model/City.java
package io.github.rajendrasatpute.samplespringmvcapi.model;

import java.sql.Timestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "cities")
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class City {
    @Id
    @Column(name = "city")
    private String cityName;
    @Column(name = "lat")
    private String latitude;
    @Column(name = "lng")
    private String longitude;
    @Column(name = "deletionTimestamp")
    private Timestamp deletionTimestamp;

    public boolean isDeleted() {
        return deletionTimestamp != null;
    }
}

File: go/sample-gin-api/model/city.go
package model

type City struct {
	CityName  string `json:"city" gorm:"column:city;type:varchar(100);primaryKey"`
	Latitude  string `json:"latitude" gorm:"column:lat;type:varchar(100)"`
	Longitude string `json:"longitude" gorm:"column:lng;type:varchar(100)"`
}

type Tabler interface {
	TableName() string
}

func (City) TableName() string {
	return "cities"
}
